R/download_package.R

```diff
@@ -0,0 +1,327 @@
+    package <- arcticdatautils::get_package(mn, resource_map_pid, file_names = TRUE)
+    child_packages <- list()
+
+    # Get child package pids
+    if (download_child_packages == TRUE) {
+        # Check that child packages exist
+        if (length(package$child_packages) != 0) {
+            n_child_packages <- length(package$child_packages)
+            progressBar <- txtProgressBar(min = 0, max = n_child_packages, style = 3)
+
+            message("\nDownloading identifiers from child packages...")
+
+            # Loop through child packages and extract pids using get_package()
+            child_packages <- lapply(seq_len(n_child_packages), function(i) {
+                setTxtProgressBar(progressBar, i)
+                arcticdatautils::get_package(mn, package$child_packages[i], file_names = TRUE)
+            })
+
+            close(progressBar)
+        }
+    }
+
+    # Initialize data pids and filename prefixes vectors
+    data_pids <- vector("character")
+    filename_prefixes <- vector("character")
+
+    # Select data pids from initial package, if they exist
+    if (length(package$data) != 0) {
+        data_pids <- package$data
+    }
+
+    # Select data pids from child packages and add to data_pids
+    child_data_pids <- unlist(lapply(child_packages, function(package) {
+        return(package$data)
+    }))
+
+    data_pids <- c(data_pids, child_data_pids)
+
+    # Check that data exists
+    if (length(data_pids) == 0) {
+        stop("No data selected.  Double check the package you entered contains data files")
+    }
+
+    # Create filename prefixes
+    all_packages <- c(package, child_packages)
+    if (prefix_file_names == TRUE) {
+        filename_prefixes <- unlist(lapply(all_packages, function(package) {
+            prefix <- pid_to_prefix(all_packages$metadata)
+            return(rep(prefix, length(all_packages$data)))
+        }))
+    }
+
+    # Check total download size
+    if (check_download_size) {
+        child_package_resource_map_pids <- lapply(child_packages, function(package) {
+            package$resource_map
+        })
+
+        downloadSize <- sum(
+            vapply(c(resource_map_pid, child_package_resource_map_pids), function(pid) {
+                get_package_size(mn, pid)
+            }, 0)
+        )
+
+        # Simplify downloadSize to readable format
+        if (downloadSize >= 1e+12) {
+            downloadSize <- round(downloadSize/(1e+12), digits = 2)
+            unit = " terabytes"
+        } else if (1e+12 > downloadSize & downloadSize >= 1e+9) {
+            downloadSize <- round(downloadSize/(1e+9), digits = 2)
+            unit = " gigabytes"
+        } else if (1e+9 > downloadSize & downloadSize >= 1e+6) {
+            downloadSize <- round(downloadSize/(1e+6), digits = 2)
+            unit = " megabytes"
+        } else if (1e+6 > downloadSize) {
+            downloadSize = round(downloadSize/1000, digits = 2)
+            unit = " kilobytes"
+        }
```

Comment:
an alternative if you want to try it: https://www.rdocumentation.org/packages/gdata/versions/2.18.0/topics/humanReadable

Reply:
Good idea but I'm not going to add it to minimize package dependencies.  I'm going to convert this to a helper function.